bench/benchmark2html.Rmd

```diff
@@ -0,0 +1,93 @@
+---
+title: '{gtsummary} benchmark'
+author: "Gustavo Zapata Wainberg & Daniel Sjoberg"
+output: html_document
+---
+
+
+Updated: `r format(Sys.Date(), "%B %d %Y")`  
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+
+library(gert)
+library(dplyr)
+# library(stringr)
+library(ggplot2)
+library(bench) # requires dev version!
+library(plotly)
+
+bench::cb_fetch(remote = "upstream")
+
+# Get the data with previous benchmarks
+stored_benchmarks <- bench::cb_read()
```

Comment:
We could add something like this to remove the duplicate benchmarks

```r
# removing duplicate checks within a single HASH (not sure why they are there)
# TODO: Learn why there are duplicates and ensure we're picking the correct one.
# My theory is that there are multiple when I was running the commands
# locally, then pushed to GH where the benchmarks were run again.
stored_benchmarks$benchmarks <-
  stored_benchmarks$benchmarks %>%
  map(
    ~arrange(.x, file, name, time) %>%
      group_by(file, name) %>%
      slice(n()) %>%
      ungroup()
  )
```

Reply:
Calling this function as you wrote it gives the following error:

Error: arrange() failed at implicit mutate() step. 
* Problem with `mutate()` input `..1`.
x Input `..1` must be a vector, not a function.
ℹ Input `..1` is `file`.
Run `rlang::last_error()` to see where the error occurred.


I've pasted your code but commented, in order to be reviewed.

Reply:
Ah ok, I just updated to this

```r
stored_benchmarks <-
  stored_benchmarks %>%
  rowwise() %>%
  filter(nrow(benchmarks) > 0) %>% # removing empty results (why are there so many empty results?)
  mutate(
    benchmarks = 
      arrange(benchmarks, file, name, time) %>%
      group_by(file, name) %>%
      slice(n()) %>%
      ungroup() %>%
      list()
  )
```

Reply:
Empty results come from my commits, in my commits the benchmark workflow won't succeed  (I guess that because of permissions to access to your github, hopefully they will succeed when they are run from your fork), so my commits add an entry to the file but without results.

Reply:
Hmmmm interesting....I wonder if there is a way to make this work for mine and yours? Ultimately we'd want to see if either my updates or another contributor's updates imrpove/slow down the package....hmmmm

Reply:
I added this to the benchmark workflow....I think the pull request portion will make the benchmark results calculate for all pull requests too

```
on:
  push:
  pull_request:
```

Reply:
Effectively it is failing due to denied permission to access

![image](https://user-images.githubusercontent.com/6965617/109435772-6eb59900-7a1c-11eb-956f-a91337e4ae71.jpeg)

Reply:
Maybe you need to change the permissions associated with your token for this action (I'm just guessing, but if I'm understanding this correctly, pull requests should need read and write access)

![image](https://user-images.githubusercontent.com/6965617/109436626-bfc78c00-7a20-11eb-9b2d-71d6b8bf532f.jpeg)

Reply:
🙄 oy, I am not sure what to do about that....I just use the standard setup I copied from some tidyverse package (I think it was dplyr or bench).

It seems that the permission is only denied on the pull request from outside accounts, but allows mine?

I think it may need to be a problem to be solved another day, and we can go ahead and move forward with this issue present?  What do you think?

Reply:
> Ah ok, I just updated to this
> 
> ```r
> stored_benchmarks <-
>   stored_benchmarks %>%
>   rowwise() %>%
>   filter(nrow(benchmarks) > 0) %>% # removing empty results (why are there so many empty results?)
>   mutate(
>     benchmarks = 
>       arrange(benchmarks, file, name, time) %>%
>       group_by(file, name) %>%
>       slice(n()) %>%
>       ungroup() %>%
>       list()
>   )
> ```

Hi!

Repeated tests results are due to repeated runs of continuous benchmarks (I know, I'm just too clever! ;) ).

In the picture you can see info about commit [89b04464](https://github.com/ddsjoberg/gtsummary/commits/master), there are two runs of continuous benchmarks (CB); do you remember if you made two pushes for the same commit? (maybe a few changes after the first commit and before the pull request was merged, they are like 3 hours apart).

Anyways, I think that in the case that there are two or more results, we should keep just the latest (mos recent) result from the benchmark data.

I'm testing some things to see if there is a way to run CB from any fork (independently of who owns it).

What others things are pending to solve? 

![image](https://user-images.githubusercontent.com/6965617/109528549-154d7880-7ab5-11eb-8f1c-d1d9f8f5ff6f.png)